plutus-contract/src/Plutus/Contract/CardanoAPITemp.hs

```diff
@@ -0,0 +1,162 @@
+{-# LANGUAGE BlockArguments           #-}
+{-# LANGUAGE DisambiguateRecordFields #-}
+{-# LANGUAGE EmptyCase                #-}
+{-# LANGUAGE FlexibleContexts         #-}
+{-# LANGUAGE GADTs                    #-}
+{-# LANGUAGE NamedFieldPuns           #-}
+{-# LANGUAGE TypeApplications         #-}
+-- Code temporarily copied over from cardano-api,
+-- until https://github.com/input-output-hk/cardano-node/pull/2936 or something similar gets merged.
+module Plutus.Contract.CardanoAPITemp (makeTransactionBody') where
+
+import qualified Data.Map.Strict                    as Map
+import qualified Data.Sequence.Strict               as Seq
+import qualified Data.Set                           as Set
+
+import           Cardano.Api
+import           Cardano.Api.Shelley                hiding (toShelleyTxOut)
+import           Cardano.Ledger.BaseTypes           (StrictMaybe (..))
+import           Cardano.Ledger.Crypto              (StandardCrypto)
+import           Ouroboros.Consensus.Shelley.Eras   (StandardAlonzo)
+
+import qualified Cardano.Ledger.Core                as Ledger
+import qualified Cardano.Ledger.Shelley.Constraints as Ledger
+
+import qualified Cardano.Ledger.Alonzo.Data         as Alonzo
+import qualified Cardano.Ledger.Alonzo.Tx           as Alonzo
+import qualified Cardano.Ledger.Alonzo.TxBody       as Alonzo
+import qualified Cardano.Ledger.Alonzo.TxWitness    as Alonzo
+
+import qualified Cardano.Ledger.ShelleyMA.TxBody    as Allegra
+
+import qualified Cardano.Ledger.Keys                as Shelley
+import qualified Shelley.Spec.Ledger.Tx             as Shelley
+import qualified Shelley.Spec.Ledger.TxBody         as Shelley
+
+makeTransactionBody' :: TxBodyContent BuildTx AlonzoEra -> Either TxBodyError (TxBody AlonzoEra)
+makeTransactionBody'
+    txbodycontent@TxBodyContent {
+        txIns,
+        txInsCollateral,
+        txOuts,
+        txFee,
+        txValidityRange = (lowerBound, upperBound),
+        txExtraScriptData,
+        txExtraKeyWits,
+        txWithdrawals,
+        txCertificates,
+        txMintValue,
+        txScriptValidity
+    } =
+    return $
+      ShelleyTxBody ShelleyBasedEraAlonzo
+        (Alonzo.TxBody
+          (Set.fromList (map (toShelleyTxIn . fst) txIns))
```

Comment:
As far as changes from the original, you just removed the guard here?

Reply:
Yes.